include/aspect/postprocess/tracer.h

```diff
@@ -35,36 +38,125 @@ namespace aspect
+         * processing.
+         *
+         * @return A pair of strings that will be printed to the screen after
+         * running the postprocessor in two columns; typically the first
+         * column contains a description of what the data is and the second
+         * contains a numerical value of this data. If there is nothing to
+         * print, simply return two empty strings.
+         */
+        virtual
+        std::pair<std::string,std::string> execute (TableHandler &statistics);
+
+        /**
+         * Save the state of this object.
+         */
+        virtual
+        void save (std::map<std::string, std::string> &status_strings) const;
+
+        /**
+         * Restore the state of the object.
+         */
+        virtual
+        void load (const std::map<std::string, std::string> &status_strings);
+
+        /**
+         * Serialize the contents of this class as far as they are not read
+         * from input parameter files.
+         */
+        template <class Archive>
+        void serialize (Archive &ar, const unsigned int version);
+
+        /**
+         * Declare the parameters this class takes through input files.
+         */
+        static
+        void
+        declare_parameters (ParameterHandler &prm);
+
+        /**
+         * Read the parameters this class declares from the parameter file.
+         */
+        virtual
+        void
+        parse_parameters (ParameterHandler &prm);
+
       private:
         /**
          * The world holding the particles
          */
-        Particle::World<dim, Particle::BaseParticle<dim> >              world;
+        Particle::World<dim>              world;
 
         /**
          * The integrator to use in moving the particles
          */
-        Particle::Integrator::Interface<dim, Particle::BaseParticle<dim> >  *integrator;
+        Particle::Integrator::Interface<dim>  *integrator;
+
+        /**
+         * The interpolator to get particle properties
+         */
+        Particle::Interpolator::Interface<dim>  *interpolator;
 
         /**
          * Pointer to an output object
          */
-        Particle::Output::Interface<dim, Particle::BaseParticle<dim> >      *output;
+        Particle::Output::Interface<dim>      *output;
 
         /**
          * Pointer to a generator object
          */
-        Particle::Generator::Interface<dim, Particle::BaseParticle<dim> >   *generator;
+        Particle::Generator::Interface<dim>   *generator;
 
         /**
-         * Whether this set has been initialized yet or not
+         * Pointer to a properties object
          */
-        bool                            initialized;
+        Particle::Property::Manager<dim>   property_manager;
 
         /**
-         * Number of initial particles to create
+         * Whether this set has been initialized yet or not
          */
-        unsigned int                    n_initial_tracers;
+        bool                            initialized;
```

Comment:
Sadly I do not see a better way. The problem here is that the tracers are currently generated within the execute() function after the first time step and therefore I need to check within this function whether to generate new tracers. I could replace this bool by a check for `global_particle_count == 0`, but what happens, if one boundary is open, and all tracers flow out of the domain? Then I would generate new tracers at some arbitrary time. A different approach could be to destroy the particle generator after generation, and then check, if the generator was destroyed. But this seems a bit drastic, and I could imagine that we at some point would like to use the generator again (e.g. for producing new particles, when some have left or they have clustered unfavourably). Therefore, I need to track if we are in this first generation timestep. Maybe we could circumvent the problem by generating the tracers within the set-up of the initial condition. This would be needed anyway, if other parts of the computation would like to use the tracers actively, but it needs some more changes, so I would suggest to leave this for now.